docs/fluent-conversion.rst

```diff
@@ -14,39 +14,74 @@ from .lang to .ftl files. This document will cover the usage of these.
 If you've got a translated page you'd like to convert as-is to the Fluent system
 then you can follow this procedure to get a big head start.
 
+The key concept in converting a page to Fluent is a recipe. The recipe is a
+piece of python code that can programmatically generate a Fluent file. It can
+use existing Fluent files as templates, and .lang localizations as data sources.
+
+All the functionality is provided by the ``fluent`` management command via
+subcommands for each phase.
+
+Create a recipe from a template
+===============================
+
+The first step is to create the recipe. Let's say you want to convert
+``bedrock/mozorg/templates/mozorg/mission.html``, then you'll run
+
+
+.. code-block:: bash
+
+    $ make run-shell
+    $ ./manage.py fluent recipe bedrock/mozorg/templates/mozorg/mission.html
+
+This will parse all of the calls to ``_()`` and the ``trans`` blocks, process the strings in the
```

Comment:
One of the reasons I got the strings from the .lang files instead of templates is that templates often include strings that are translated in other files, e.g. a global file like newsletter.lang. Extracting everything in the template will result in string IDs with no translations in the .lang files you're porting.

Reply:
https://github.com/mozilla/bedrock/blob/712b92ec732c8e9708bd63443b1c73fc9dba067e/lib/l10n_utils/management/commands/_fluent_recipe.py#L118-L129 keeps track of which string is coming from which lang file.

One nice thing about the migration being the first thing is that it'll create a migration for each file found. If that touches more .lang files than you want right now, you just remove the `add_transforms` call for that ftl file from the recipe, and the next steps will only bother about the content you left in the recipe.

Well, it doesn't do the default ones yet, not sure how to tackle those yet.